cudaaligner/CMakeLists.txt

```diff
@@ -38,8 +38,15 @@ add_library(cudaaligner_internal INTERFACE)
 target_include_directories(cudaaligner_internal INTERFACE
         ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
+if(cga_enable_caching_allocator)
+        if (cga_enable_preallocating_caching_allocator)
+                target_compile_definitions(cudaaligner PUBLIC CGA_ENABLE_CACHING_ALLOCATOR CGA_ENABLE_PREALLOCATING_ALLOCATOR)
+        else()
+                target_compile_definitions(cudaaligner PUBLIC CGA_ENABLE_CACHING_ALLOCATOR)
+        endif()
+endif()
 
-target_link_libraries(cudaaligner cgautils cgalogging)
+target_link_libraries(cudaaligner cgautils cgalogging cub)
```

Comment:
CUB is only needed if `cga_enable_caching_allocator` is true and `cga_enable_preallocating_caching_allocator` is false, right?
(we can also leave it is it is, we anyway get CUB)

Reply:
Yes, you are right, but if you are ok with it, I'd leave it as it is, because it is pulled anyways. It keeps the `CMakeLists.txt` simple.

Reply:
Yes, I absolutely agree